dashboard/public/js/tutorial.js

```diff
@@ -0,0 +1,368 @@
+// Tutorial handling
+
+function sugarizerTour(currentView, role) {
+	console.log("sugarizerTour", currentView, role);
+	var tutorial = {};
+	var tour;
+	var launched = false;
+    
+	tutorial.elements = [];
+
+	// Init tutorial
+	tutorial.init = function() {
+		var prevString = "Prev";
+		var nextString = "Next";
+		var endString = "End";
+		tour = new window.Tour({
+			name: currentView,
+			template: "\
+			<div class='popover tour'>\
+				<div class='arrow'></div>\
+				<h3 class='popover-title tutorial-title'></h3>\
+				<table><tr><td style='vertical-align:top;'><div id='icon-tutorial' style='visibility:hidden;display:inline-block;'></div>\
+				</td><td><div class='popover-content'></div></td></tr></table>\
+				<div class='popover-navigation' style='display: flex; flex-wrap:wrap; justify-content: center; align-items: center'>\
+					<div class='tutorial-prev-icon icon-button' data-role='prev'>\
+						<div class='tutorial-prev-icon1 web-activity'>\
+							<div class='tutorial-prev-icon2 web-activity-icon'></div>\
+							<div class='tutorial-prev-icon3 web-activity-disable'></div>\
+						</div>\
+						<div class='icon-tutorial-text'>"+prevString+"</div>\
+					</div>\
+					<span data-role='separator' style='margin: 4px'>|</span>\
+					<div class='tutorial-next-icon icon-button' data-role='next'>\
+						<div class='tutorial-next-icon1 web-activity'>\
+							<div class='tutorial-next-icon2 web-activity-icon'></div>\
+							<div class='tutorial-next-icon3 web-activity-disable'></div>\
+						</div>\
+						<div class='icon-tutorial-text'>"+nextString+"</div>\
+					</div>\
+					<div class='tutorial-end-icon icon-button' data-role='end'>\
+						<div class='tutorial-end-icon1 web-activity'>\
+							<div class='tutorial-end-icon2 web-activity-icon'></div>\
+							<div class='tutorial-end-icon3 web-activity-disable'></div>\
+						</div>\
+						<div class='icon-tutorial-text'>"+endString+"</div>\
+					</div>\
+				</div>\
+			</div>",
+			storage: window.localStorage,
+			backdrop: true,
+			autoscroll: true,
+			steps: [],
+			keyboard: true,
+			onEnd: function() {
+				tutorial.elements = [];
+			}
+		});
+		if (currentView == "home") {
+			tour.addSteps([
+				{
+					element: "",
```

Comment:
Quick feedback:
- put steps for each module in a separate file. e.g. `steps/home.json`
- how are you handling multiple languages.

Reply:
I'll be using webl10n for translations. I was following Sugarizer tutorial format (https://github.com/llaske/sugarizer/blob/dev/lib/tutorial.js), that why I put it all in a single file.